osfoffline/polling_osf_manager/polling_events.py

```diff
@@ -21,117 +21,150 @@ def run(self):
 
 class CreateFolder(PollingEvent):
     def __init__(self, path):
-        super().__init__(path)
-        self.path = ProperPath(path, is_dir=True)
-        assert self.path
+        super().__init__()
+        self.path = path
 
     @asyncio.coroutine
     def run(self):
         # create local node folder on filesystem
-        if not os.path.exists(self.path.full_path):
-            AlertHandler.info(self.path.name, AlertHandler.DOWNLOAD)
-            os.makedirs(self.path.full_path)
+        try:
+            folder_to_create = ProperPath(self.path, is_dir=True)
```

Comment:
Proper path does not perform OS-specific checks. It is possible that the program could try to create a file or folder with special characters (like `< > &`), and this error handler would not catch those problems.

Some error handling may need to be added at the actual act of creating the file/folder (eg line 38). This should be tested on MS Windows by trying to sync an OSF project with an oddly named file. I can provide a preconfigured VM next time we're in airdrop range, if that would save you time.